### PR TITLE
[KEYCLOAK-3875] - Conditional OTP Forms not working as expected

### PR DESCRIPTION
**Pull request by abstractj**
We already have integration tests for it, that's the reason why it wasn't added. If someone would like to test it manually, please import these files: https://gist.github.com/abstractj/11436442080ac4c76a82785741c99f3d

### CODE REVIEW

File: services/src/main/java/org/keycloak/authentication/authenticators/browser/ConditionalOtpFormAuthenticator.java [hunk 1]
```diff
@@ -266,4 +269,22 @@ private boolean userHasRole(AuthenticationFlowContext context, String roleName)
...
+                return hasRole(user.getRoleMappings(), role);
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void setRequiredActions(KeycloakSession session, RealmModel realm, UserModel user) {
+        if(isOTPRequired(realm, user) && !user.getRequiredActions().contains(UserModel.RequiredAction.CONFIGURE_TOTP.name())) {
+            user.addRequiredAction(UserModel.RequiredAction.CONFIGURE_TOTP.name());
```
Comment (stianst) [inline services/src/main/java/org/keycloak/authentication/authenticators/browser/ConditionalOtpFormAuthenticator.java]: What if the config is that a user can access without OTP on an internal machine (based on header), but not remotely. Then he by accident tries to login from a remote machine and is displayed OTP form? As the required action is added to the user here he'll have to setup OTP on the internal machine as well. It should be possible to add the action to the client session instead which would prevent it from being required on a new session.

**Comment by stianst** [comment]
I'd say the current tests must be lacking or broken as they didn't catch the problems?

**Comment by abstractj** [comment]
@stianst I think the latest changes fixes the bugs in the testsuite and also for conditional OTP.  But it seems like Travis is unhappy with `The job exceeded the maximum time limit for jobs, and has been terminated`.

I believe this is not related with this PR

**Comment by abstractj** [comment]
all :green_heart:  now!

**Pull request merged**